src/arrayMethodJoin.js

```diff
@@ -5,8 +5,19 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strInsteadArray = '';
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        strInsteadArray += '';
```

Comment:
it is unnecessary to add empty string

Reply:
I need this thing ), without +='' it didn't pass the test 
"for [false, null, 1, undefined, 0, NaN, ''].join(',')"

Reply:
No, you need to **not add** undefined or null to result, not **add empty string** to result
```javascript
      if (this[i] !== undefined && this[i] !== null) {
        strInsteadArray += '' + this[i];
      }
```

Reply:
Thank you), at least I've got it)